tests/test_std.rs

```diff
@@ -575,6 +576,17 @@ fn combinations() {
         vec![3, 4],
         ]);
 
+    let it = vec![1, 2, 2, 3, 4].into_iter().unique_combinations(3);
```

Comment:
Should we assert that `unique_combinations` yields the "same" (modulo ordering) elements that would be obtained by manually filtering elements from `combinations`?

Reply:
I think that this will not always be true, as combinations explore all combinations before getting a new number for the iterator. We see all the numbers as fast as possible as we want to find the end. This is why the `LasyBuffer` is used. to not need that many next calls from the iterator.

Reply:
I am not sure if I understand you correctly. But shouldn't `unique_combinations` give the same elements that you would get by manually removing duplicates from `combinations` (possibly sorting the yielded permutations)?

Reply:
This is true as long as the iterator that combinations is done over is sorted in the first place. Might be a good test with quicksort.

Reply:
After some thinking is think we should make thise, however without garanteeing the order of the items that comes out in either cases. So sort bought on input and output. Will implement that this weekend